cwf/gateway/services/uesim/servicers/ue_store_utils.go

```diff
@@ -14,32 +14,33 @@
 package servicers
 
 import (
+	"fmt"
+
 	cwfprotos "magma/cwf/cloud/go/protos"
 	"magma/orc8r/cloud/go/blobstore"
 	"magma/orc8r/lib/go/protos"
 
-	"github.com/pkg/errors"
+	"github.com/hashicorp/go-multierror"
 )
 
 func addUeToStore(srvstore blobstore.StoreFactory, ue *cwfprotos.UEConfig) {
 	blob, err := ueToBlob(ue)
 	store, err := srvstore.StartTransaction(nil)
 	if err != nil {
-		err = errors.Wrap(err, "Error while starting transaction")
+		err = fmt.Errorf("Error while starting transaction: %w", err)
 		err = ConvertStorageErrorToGrpcStatus(err)
 		return
 	}
 	defer func() {
-		switch err {
-		case nil:
+		if err == nil {
 			if commitErr := store.Commit(); commitErr != nil {
-				err = errors.Wrap(err, "Error while committing transaction")
+				err = fmt.Errorf("Error while committing transaction: %w", commitErr)
 				err = ConvertStorageErrorToGrpcStatus(err)
 			}
-		default:
+		} else {
 			if rollbackErr := store.Rollback(); rollbackErr != nil {
-				err = errors.Wrap(err, "Error while rolling back transaction")
-				err = ConvertStorageErrorToGrpcStatus(err)
+				errs := multierror.Append(err, fmt.Errorf("Error while rolling back transaction: %w", rollbackErr))
+				err = ConvertStorageErrorToGrpcStatus(errs.ErrorOrNil())
```

Comment:
Is there a reason to use multierror in one case but not in the other? And for using multierror at all?

There is a similar case in cwf/gateway/services/uesim/servicers/uesim.go.

Reply:
If `err` is nil, but the commit fails, there is only one error to be passed on. If `err` is not nil, the it tries to roll back, but if that also fails, we have two errors to pass: `err` and `rollbackErr`, that's why we use a multierror here.

We need to pass `commitErr` , because `err` is nil by default here due to the `if` clause, otherwise we would try to pass an error, but the error we pass is nil, so in the end the error gets ignored.
In the other case we use the multierror to pass on both, `err` and `rollbackErr`.

Reply:
I agree with @wolfseb

Reply:
Ah, so this goes beyond replacing pkg/errors and fixes broken error handling. Probably should mention that in the summary as well.

Reply:
I added this to the summary.